clang/lib/DPCT/Diagnostics.h

```diff
@@ -92,7 +117,7 @@ enum class Warnings {
 #undef DEF_ERROR
 #undef DEF_WARNING
 #undef DEF_COMMENT
-  END
+  END = MaxWarningID + 1
```

Comment:
How about remove `BEGIN` and `END`, use static variable in class `DiagnosticsMessage` for the same functionality.